### PR TITLE
[10.0] migrate base_gs1_barcode

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**

### CODE REVIEW

**Comment by colmeirin** [comment]
Hello, does this work as a base for a 9.0 migration?
Regards

File: base_gs1_barcode/__manifest__.py
Comment (alexis-via) [inline base_gs1_barcode/__manifest__.py]: Do you have the agreement of Numérigraphe to change the licence from GPL to AGPL ?

File: base_gs1_barcode/models/gs1_barcode.py
Comment (alexis-via) [inline base_gs1_barcode/models/gs1_barcode.py]: You can do : from odoo import api, fields, models, _
It saves one line :)

File: base_gs1_barcode/views/gs1_barcode_view.xml
Comment (alexis-via) [inline base_gs1_barcode/views/gs1_barcode_view.xml]: what about the parent parameter ?

**Review by alexis-via — APPROVED** [review]

**Comment by alexis-via** [comment]
FYI, the date parsing when the day = 00 is wrong: https://www.gs1.org/docs/barcodes/GS1_General_Specifications.pdf page 143 section 3.4.7 first note:
```
Note : When it is not necessary to specify the day (the day field is filled with two zeros), the 
resultant data string SHALL be interpreted as the last day of the noted month including any 
adjustment for leap years (e.g., “130200” is “2013 February 28”, “160200” is “2016 February 
29”, etc.).
```
It's a small detail, but...

File: base_gs1_barcode/__manifest__.py
Comment (legalsylvain) [inline base_gs1_barcode/__manifest__.py:12]: could you move all the description into the readme file ?

File: base_gs1_barcode/test/gs1_barcode_test.yml
Comment (legalsylvain) [inline base_gs1_barcode/test/gs1_barcode_test.yml]: why did you removed the test ?

File: base_gs1_barcode/__manifest__.py
Comment (gurneyalex) [reply base_gs1_barcode/__manifest__.py:12]: yes please

**Review by gurneyalex — CHANGES_REQUESTED** [review]
Please revert licence change (unless you have an agreement from all contributors that they agree).

**Comment by alexis-via** [comment]
FYI, I made a PR to acsone's branch to update this PR https://github.com/acsone/stock-logistics-barcode/pull/2

**Comment by Cedric-Pigeon** [comment]
@alexis-via I merged your PR. thanks a lot

**Review by alexis-via — APPROVED** [review]

**Comment by alexis-via** [comment]
@gurneyalex licence change has been reverted ; could you remove your "change request" ?

**Review by rafaelbn — APPROVED** [review]

**Comment by rafaelbn** [comment]
Please @Cedric-Pigeon check travis, is :red_circle:

**Comment by rousseldenis** [comment]
@numerigraphe @clonedagain Do you see objection to AGPL change ?

**Comment by clonedagain** [comment]
@rousseldenis sorry your message had not reached me until today.
I need to ask the boss before I can agree officially.
I'll get back to you soon.

**Comment by rousseldenis** [comment]
@clonedagain No problem. This is to have a clear answer on that point before merging.

**Comment by clonedagain** [comment]
@rousseldenis sorry for the delay about the re-licensing. The person in charge of IP admits its a subtle change and he needs time to consider it.
Maybe someone from OCA can give us a call (in French if at all possible), if you really need our approval ?

That being said, doesn't the OCA CLA give you power to proceed with the change already ?

**Comment by rousseldenis** [comment]
@clonedagain It was just to know if we wait before merge or merge like this

**Pull request closed without merging**
